sdk/src/bank_hash.rs

```diff
@@ -0,0 +1,163 @@
+use crate::hash::Hash;
+use bincode::deserialize_from;
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaChaRng;
+use serde::{Deserialize, Serialize, Serializer};
+use std::fmt;
+use std::io::Cursor;
+
+// Type for representing a bank accounts state.
+// Taken by xor of a sha256 of accounts state for lower 32-bytes, and
+// then generating the rest of the bytes with a chacha rng init'ed with that state.
```

Comment:
Is there a citation for using the RNG like this?  It seems like we should seed the RNG with the account hash, then generate the _entire_ bank hash from it.  Otherwise, I think an attacker can discard the extra bytes and work the problem as if it were on the bare SHA256 hashes.

Reply:
Why can they discard? They will still have to match those rng bits as well to make an attack otherwise their final image value will not hash to the same value as the real one.

Reply:
Maybe I'm missing something.  Do the RNG bits get mixed with the SHA256 bits somewhere?

My thinking is that data of type `[[SHA256][CHACHA20_EXTEND]]` can be XORed as many times as one likes and the SHA256 and CHACHA20_EXTEND components individually will be the same, so can be treated as independent.

Reply:
The RNG bits are seeded from the sha256 bits.

Reply:
OK, right I had my head stuck on the second pre-image case, where only one account hash would be replaced from a known.  I think this should be fine against finding a new basket of account states that match the expected hash